app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -98,12 +103,22 @@ fun KeyboardScreen(
                 ?: DEFAULT_KEYBOARD_LAYOUT,
         ].keyboardDefinition
 
-    val keyboard =
+    val (keyboard, secondaryKeyboard, tertiaryKeyboard) =
         when (mode) {
-            KeyboardMode.MAIN -> keyboardDefinition.modes.main
-            KeyboardMode.SHIFTED -> keyboardDefinition.modes.shifted
-            KeyboardMode.NUMERIC -> keyboardDefinition.modes.numeric
-            else -> KB_EN_THUMBKEY_MAIN
+            KeyboardMode.MAIN ->
```

Comment:
This is so strange because you already have the map with the 3 layouts, why do we need to do this for the full layouts again.

Reply:
This should be resolved now with the latest commit, I moved this logic to the arguments of `KeyboardKey` instead